docs/current_state_specification.md

```diff
@@ -0,0 +1,153 @@
+### Current State Representation
+
+The cloud backend (or any OTA update system) needs to keep track of the current state of the device. This current state information is relevant as input for the context pipeline for compiling future desired states and acts as an indicator of problems in applying a given desired state, if the current state differs from the desired state, which may not be reached.
+
+For representing the current state of the device, the approach of a holistic inventory of hardware and software nodes will be applied. This model is using a graph structure consisting of three element types - hardware nodes, software nodes, associations.
+
+### Current State Data Model
+
+The following table describes all supported properties and sections of the Current State specification:
+
+| Property | Type | Description |
+| - | - | - |
+| **General properties** | | |
+| hardwareNodes | JSON array | Inventory for a list of hardware nodes |
+| softwareNodes | JSON array | Inventory for a list of software nodes |
+| associations | JSON array | List of mappings between the inventory nodes |
+| **Hardware node properties** | | |
+| id | string | Identifier of the hardware node |
+| version | string | Version of the hardware node |
+| name | string | Name of a hardware node |
+| parameters | JSON array | List of key/value parameters for a hardware node |
+| addressable | boolean | Enables hardware node addressability |

```

Comment:
What do you mean exactly by addressable? 
Are we talking about network connectivity and if yes on which layer of the OSI model?

Reply:
To be honest, I don't know what exactly addressable means, for the moment we treat is just as a reserved property in the structure, but currently not really used inside the Kanto project.
For the current state specification we did not want to come up with something brand new, but tried to make use of an existing concept. Actually we borrowed the Inventory definitions from another project [Eclipse BfB](https://github.com/eclipse-bfb). Unfortunately, this project has not produced anything yet within its repositories and nothing is public there yet.
Let's look at the implementation inside the Kanto Update manager as a preview. More details can provided by the Eclipse BfB Project Lead - [Ioannis Spyropoulos](mailto:Ioannis.Spyropoulos@etas.com) from Bosch/ETAS.

Reply:
I See, maybe the maintainers of the Eclipse BfB can comment on this.